File: c/002.c
#include <stdio.h>

int main()
{
  int max = 4000000;

  int total = 0;
  int previous = 1;
  int current = 1;
  while (current < max)
  {
    int next = current + previous;
    if (!(current % 2))
      total += current;
    previous = current;
    current = next;
  }
  printf("%d\n", total);
  return(0);
}

File: perl6/002.pl
#!/usr/bin/perl6 -w
#use strict;

my $total = 0;
my $max = 4000000;
my $previous = 1;
my $current = 1;
while ($current < $max) {
  my $next = $previous + $current;
  $total += $current if (!($current % 2));
  $previous = $current;
  $current = $next;
}
say $total;
